widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx

```diff
@@ -0,0 +1,220 @@
+import type { Mode } from '../../store/config';
+import type { WidgetColorsKeys } from '@rango-dev/widget-embedded';
+
+import {
+  ChevronDownIcon,
+  ChevronUpIcon,
+  Collapsible,
+  CustomColorsIcon,
+  Divider,
+  Typography,
+} from '@rango-dev/ui';
+import React, { useEffect, useState } from 'react';
+
+import { ColorItemPicker } from '../../components/ColorItemPicker';
+import { WIDGET_COLORS } from '../../constants';
+import { useConfigStore } from '../../store/config';
+
+import {
+  Color,
+  ColorsContent,
+  CustomColorCollapsible,
+  CustomColors,
+  FieldTitle,
+  Row,
+} from './StyleLayout.styles';
+
+const Colors = (props: { mainColor?: string; secondColor?: string }) => {
+  return (
+    <ColorsContent>
+      <Color style={{ backgroundColor: props.mainColor }} />
+      {props.secondColor !== undefined && (
+        <Color
+          style={{
+            backgroundColor: props.secondColor,
+            position: 'absolute',
+            right: -10,
+          }}
+        />
+      )}
+    </ColorsContent>
+  );
+};
+
+export function CustomColorsSection(props: CustomColorsTypes) {
+  const { tab, selectedPresets } = props;
+  const [openCustomColors, toggleCustomColors] = useState<{
+    tab: Mode;
+    value: boolean;
+  }>({ tab, value: false });
+  const [openCustomColor, setOpenCustomColor] = useState<string | null>(null);
+  const onChangeColors = useConfigStore.use.onChangeColors();
+
+  const { theme } = useConfigStore.use.config();
+  const isDarkTab = tab === 'dark';
+  const isLightTab = tab === 'light';
+  const isAutoTab = tab === 'auto';
+  const isDarkColorsOnly =
+    !!Object.keys(theme?.colors?.dark).length &&
+    !Object.keys(theme?.colors?.light).length;
+  const isLightColorsOnly =
+    !!Object.keys(theme?.colors?.light).length &&
+    !Object.keys(theme?.colors?.dark).length;
+  const isLightAndDarkColors =
+    !!Object.keys(theme?.colors?.light).length &&
+    !!Object.keys(theme?.colors?.dark).length;
+
+  const isOpenCustomColors =
+    tab === openCustomColors.tab && openCustomColors.value;
+
+  const handleOpenCollapse = (key: string) => {
+    if (openCustomColor === key) {
+      setOpenCustomColor(null);
+    } else {
+      setOpenCustomColor(key);
+    }
+  };
+  useEffect(() => {
+    setOpenCustomColor(null);
+  }, [tab]);
+
+  const onResetColor = (name: WidgetColorsKeys, mode: 'light' | 'dark') => {
+    const color = !!selectedPresets ? selectedPresets[mode][name] : undefined;
+    onChangeColors(name, mode, color);
+  };
+
+  const getMainColor = (
+    key: WidgetColorsKeys,
+    mode?: 'light' | 'dark'
+  ): string | undefined => {
+    if (isAutoTab && isLightAndDarkColors) {
+      return theme?.colors[mode || 'light'][key];
+    }
+    if (isDarkTab && isDarkColorsOnly) {
+      return theme?.colors?.dark[key];
+    }
+    if (isLightTab && isLightColorsOnly) {
+      return theme?.colors?.light[key];
+    }
+    return undefined;
+  };
+
+  return (
+    <Collapsible
```

Comment:
Also, their heights are not matched. 
![image](https://github.com/rango-exchange/rango-client/assets/123297941/aeecda8f-4b74-4593-9736-e60ff143b3c8)


I think the width of the whole panel should be wider. It might be related to the main Collapsible component.